diff --git a/node_modules/@fingerprintjs/fingerprintjs-pro-react-native/android/src/main/java/com/fingerprintjs/reactnative/RNFingerprintjsProModule.kt b/node_modules/@fingerprintjs/fingerprintjs-pro-react-native/android/src/main/java/com/fingerprintjs/reactnative/RNFingerprintjsProModule.kt
index c620fc0..a3a91b5 100644
--- a/node_modules/@fingerprintjs/fingerprintjs-pro-react-native/android/src/main/java/com/fingerprintjs/reactnative/RNFingerprintjsProModule.kt
+++ b/node_modules/@fingerprintjs/fingerprintjs-pro-react-native/android/src/main/java/com/fingerprintjs/reactnative/RNFingerprintjsProModule.kt
@@ -56,7 +56,8 @@ class RNFingerprintjsProModule(reactContext: ReactApplicationContext) : ReactCon
   @ReactMethod
   fun getVisitorId(tags: ReadableMap?, linkedId: String?, promise: Promise) {
     try {
-      fpjsClient?.getVisitorId(tags?.toHashMap() ?: emptyMap(),
+      @Suppress("UNCHECKED_CAST")
+      fpjsClient?.getVisitorId((tags?.toHashMap() ?: emptyMap()) as Map<String, Any>,
         linkedId ?: "",
         { result -> promise.resolve(result.visitorId) },
         { error -> promise.reject("Error: ", getErrorDescription(error))
@@ -69,7 +70,8 @@ class RNFingerprintjsProModule(reactContext: ReactApplicationContext) : ReactCon
   @ReactMethod
   fun getVisitorData(tags: ReadableMap?, linkedId: String?, promise: Promise) {
     try {
-      fpjsClient?.getVisitorId(tags?.toHashMap() ?: emptyMap(),
+      @Suppress("UNCHECKED_CAST")
+      fpjsClient?.getVisitorId((tags?.toHashMap() ?: emptyMap()) as Map<String, Any>,
         linkedId ?: "",
         { result -> promise.resolve(Arguments.fromList(listOf(result.requestId, result.confidenceScore.score, result.asJson))) },
         { error -> promise.reject("Error: ", getErrorDescription(error))
